src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/TraceLogging/TraceLoggingEventSource.cs

```diff
@@ -378,7 +378,9 @@ public unsafe void Write(string? eventName, EventSourceOptions options)
 
             if (this.IsEnabled((EventLevel)level, keywords))
             {
+                m_activeWritesCount++;
                 WriteMultiMergeInner(eventName, ref options, eventTypes, activityID, childActivityID, values);
+                m_activeWritesCount--;
```

Comment:
Why are the other uses wrapped in finally but not this one?

Reply:
The ones in finally blocks are there because those `Write` methods had pre-existing `try`/`catch` blocks in them. These just didn't happen to. To prevent unpaired incr/decrs I should wrap them all in `try`/`catch`/`finally` blocks.